### ISSUE TITLE
Okta & AWS login rule false positives for failed logins

### ISSUE THREAD

**Issue by maxbfigma** [panther-labs/panther-analysis#131]
We've observed a number of cases where our Panther instance sent alerts for failed login attempts that were not actionable. I believe these rules should be updated in order to only fire alerts when a login is successful. This will increase their trustworthiness and reduce the likelihood of alert fatigue.

## AWS
The `AWS.Root.Activity` rule will trigger for some failed login attempts, setting off a high-priority alert for what is presumably a low-complexity brute-force bot.

Example (redacted) event that triggers this:
```
{
	"p_rule_id": "AWS.Root.Activity",
	"p_rule_tags": [
		"AWS",
		"Identity & Access Management"
	],
	"p_rule_reports": {
		"CIS": [
			"3.3"
		]
	},
	"p_alert_id": "redacted",
	"p_alert_creation_time": "redacted",
	"p_alert_update_time": "redacted",
	"p_rule_error": null,
	"awsRegion": "redacted",
	"errorMessage": "Not a valid response for the provided request id: aws_ccV60redacted",
	"eventID": "redacted",
	"eventName": "ExternalIdPDirectoryLogin",
	"eventSource": "signin.amazonaws.com",
	"eventTime": "redacted",
	"eventType": "AwsConsoleSignIn",
	"eventVersion": "1.05",
	"readOnly": false,
	"recipientAccountId": "redacted",
	"requestID": "redacted",
	"responseElements": {
		"ExternalIdPDirectoryLogin": "Failure"
	},
	"sourceIPAddress": "redacted",
	"userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.105 Safari/537.36",
	"userIdentity": {
		"type": "Root",
		"principalId": "redacted",
		"arn": "",
		"accountId": "redacted",
		"accessKeyId": ""
	},
	"p_log_type": "AWS.CloudTrail",
	"p_row_id": "redacted",
	"p_event_time": "redacted",
	"p_parse_time": "redacted",
	"p_source_id": "redacted",
	"p_source_label": "CloudTrail",
	"p_any_ip_addresses": [
		"redacted"
	],
	"p_any_aws_account_ids": [
		"redacted"
	],
	"p_any_aws_arns": []
}
```

## Okta
The `Okta.GeographicallyImprobableAccess` rule will trigger even for unsuccessful logins, meaning it can easily be set off by various scrapers trying to login as `admin` from various parts of the world.

Example (redacted) event that triggers this:
```
{
	"p_rule_id": "Okta.GeographicallyImprobableAccess",
	"p_rule_tags": [
		"Identity & Access Management"
	],
	"p_rule_reports": {},
	"p_alert_id": "redacted",
	"p_alert_creation_time": "redacted",
	"p_alert_update_time": "redacted",
	"p_rule_error": null,
	"uuid": "redacted",
	"published": "redacted",
	"eventType": "user.session.start",
	"version": "0",
	"severity": "INFO",
	"legacyEventType": "core.user_auth.login_failed",
	"displayMessage": "User login to Okta",
	"actor": {
		"id": "unknown",
		"type": "User",
		"alternateId": "admin",
		"displayName": "unknown"
	},
	"client": {
		"userAgent": {
			"browser": "CHROME",
			"os": "Linux",
			"rawUserAgent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.135 Safari/537.36"
		},
		"geographicalContext": {
			"geolocation": {
				"lat": -29.6116,
				"lon": -51.0933
			},
			"city": "Dois Irmaos",
			"state": "Rio Grande do Sul",
			"country": "Brazil",
			"postalCode": "93950"
		},
		"zone": "null",
		"ipAddress": "redacted",
		"device": "Computer"
	},
	"request": {
		"ipChain": [
			{
				"ip": "redacted",
				"geographicalContext": {
					"geolocation": {
						"lat": -29.6116,
						"lon": -51.0933
					},
					"city": "Dois Irmaos",
					"state": "Rio Grande do Sul",
					"country": "Brazil",
					"postalCode": "93950"
				},
				"version": "V4"
			}
		]
	},
	"outcome": {
		"result": "FAILURE",
		"reason": "VERIFICATION_ERROR"
	},
	"transaction": {
		"id": "redacted",
		"type": "WEB",
		"detail": {}
	},
	"debugContext": {
		"debugData": {
			"threatSuspected": "false",
			"url": "redacted",
			"loginResult": "VERIFICATION_ERROR",
			"requestId": "redacted",
			"requestUri": "redacted"
		}
	},
	"authenticationContext": {
		"authenticationStep": 0,
		"externalSessionId": "unknown"
	},
	"securityContext": {
		"asNumber": 263297,
		"asOrg": "renovare telecom",
		"isp": "renovare telecom",
		"domain": "rnvtelecom.com.br",
		"isProxy": false
	},
	"p_log_type": "Okta.SystemLog",
	"p_row_id": "redacted",
	"p_event_time": "redacted",
	"p_parse_time": "redacted",
	"p_source_id": "redacted",
	"p_source_label": "Okta",
	"p_any_ip_addresses": [
		"redacted"
	],
	"p_any_domain_names": [
		"rnvtelecom.com.br"
	]
}
```
Note that `event['outcome']['result']` is `FAILURE`.

**Comment by nhakmiller** [comment]
Yeah this seems reasonable, let's see about getting this in for 1.11. Should be straightforward.

**Comment by nhakmiller** [comment]
In the first case, I think the easiest general purpose solution is to not return an alert if any error message was returned (if an error message was returned the task didn't get carried out).

In the latter we can just explicitly check for failures.

I opened #145 to address this, and I used your example events here as unit tests (I hope that's ok, I can get different/new ones if it's an issue).
